Add Vale in VSCode steps (#320)
grafana/writers-toolkit: docs/sources/review/lint-prose/index.md

<!-- vale Grafana.Quotes = YES -->
```

## Use Vale in VS Code

You can use Vale to lint your current document in VS Code.

1. Clone the [Technical Documentation](https://github.com/grafana/technical-documentation) repository.

   ```bash
   git clone git@github.com:grafana/technical-documentation.git
   ```

1. Download [Vale](https://github.com/errata-ai/vale/releases).

> 1. Extract the downloaded zip file to `$HOME/vale`.

Reviewer: I might be an idiot, but I don't know how to do this. I've gone into my top-level directory (AFAIK) on my machine and displayed hidden files and I've tried doing a search using terminal for a vale file, but I don't know where this should be.

Author: Fixed this up by adding macOS specific instructions

<!-- vale Grafana.Quotes = YES -->
```

## Use Vale in VS Code

You can use Vale to lint your current document in VS Code.

1. Clone the [Technical Documentation](https://github.com/grafana/technical-documentation) repository.

   ```bash
   git clone git@github.com:grafana/technical-documentation.git
   ```

> 1. Download [Vale](https://github.com/errata-ai/vale/releases).

Reviewer: I'm not sure if this got lost in the fray or if you intended to remove this comment, but I definitely stared at this page not knowing what to download for a while. I don't think it hurts to point out that folks should look at the Assets list and find the one that's for them (already a bit of a step since I had to do further Googling to figure out which one was good for my Mac).

> 1. Download [Vale](https://github.com/errata-ai/vale/releases).

Author: It's kinda both? You download the executable.. there's no real "install step. Even brew just pulls in the executable and places it in the Cellar (although it also adds symlinks in places like `/usr/local/bin` to make it easier for users to find/use it.

> 1. Download [Vale](https://github.com/errata-ai/vale/releases).

Author: ```suggestion
1. Download and install [Vale](https://github.com/errata-ai/vale/releases).
```

> 1. Download [Vale](https://github.com/errata-ai/vale/releases).

Reviewer: I like the change to **Download and install**!

   mkdir bin && tar -xvzf vale_2.28.0_Linux_64-bit.tar.gz -C bin
   export PATH=./bin:"$PATH"
   ```

   ```macos
   brew install vale
   ```

   {{< /code >}}

1. Create a `vale.ini` file in your home directory or in a working directory with the following contents:

   ```bash
   StylesPath = /FULL_PATH_TO_REPO/technical-documentation/linters/vale
   MinAlertLevel = suggestion

   [*.md]
   BasedOnStyles = Grafana
   ```

   Replace `FULL_PATH_TO_REPO` with the full path to the cloned Technical Documentation repository. For example, in Linux you could set StylesPath to `/home/username/git-repos/technical-documentation/linters/vale` and in macOS, you could set it to `/Users/username/git-repos/technical-documentation/linters/vale`. The path depends on where you cloned the git repository.

1. Install the [Vale VS Code extension](https://marketplace.visualstudio.com/items?itemName=chrischinchilla.vale-vscode) in VS Code.

   1. Start VS Code.
   1. Press Ctrl+P, paste the following command, and press Enter.

   ```
   ext install ChrisChinchilla.vale-vscode
   ```

1. Configure the Vale VS Code extension.

>    1. Press Ctrl+Shift+X and select the Vale VS Code extension.

Reviewer: Wondering if we can add the UI navigation here as well. I'm not sure the keyboard commands worked for me. When I tried them later, I couldn't get it to work.

```suggestion
   1. Press Ctrl+Shift+X or click the **Extensions** icon and select the Vale VS Code extension.
```

   1. Select the gear icon.
   1. Set `Vale › Vale CLI: Config` to the path to your `vale.ini` file. For example, on Linux that could be `/home/USERNAME/vale.ini` and on macOS, that could be `/Users/USERNAME/vale.ini`. The path depends on where you created the `vale.ini` file.

>    1. Set `Vale › Vale CLI: Path` to the path for the vale executable. For example, in Linux, that could be `/home/USERNAME/bin/vale` and on macOS, that could be `/usr/local/bin/vale`.

Reviewer: How does one determine this just in case this isn't correct?

Author: It really depends on how you install it - the challenge of a single stand-alone executable. 

In Linux, you can download the zip file.. and put it anywhere you want/ The steps for Linux suggest putting it in `/home/USERNAME/bin`. 

In macOS, it also depends... if you use brew, it'll probably be OK to link it `/usr/local/bin/vale` since that's supposed to be a default location on macOS... but it depends on local brew config.  I'm not sure if using `which vale` will report the path to the executable... this works on Linux.

>    1. Set `Vale › Vale CLI: Config` to the path to your `vale.ini` file. For example, on Linux that could be `/home/USERNAME/vale.ini` and on macOS, that could be `/Users/USERNAME/vale.ini`. The path depends on where you created the `vale.ini` file.
>    1. Set `Vale › Vale CLI: Path` to the path for the vale executable. For example, in Linux, that could be `/home/USERNAME/bin/vale` and on macOS, that could be `/usr/local/bin/vale`.

Reviewer: Styling for UI text differentiates it from the other actual code
```suggestion
   1. Set **Vale › Vale CLI: Config** to the path to your `vale.ini` file. For example, on Linux that could be `/home/USERNAME/vale.ini` and on macOS, that could be `/Users/USERNAME/vale.ini`. The path depends on where you created the `vale.ini` file.
   1. Set **Vale › Vale CLI: Path** to the path for the vale executable. For example, in Linux, that could be `/home/USERNAME/bin/vale` and on macOS, that could be `/usr/local/bin/vale`.
```

1. Restart VS Code.

Vale lints your current document every time you save your changes. The extension reports the linting results in two ways:

> 1. In-line edit marks. You can hover your mouse cursor over the edit marks to view the vale warning or error.

Reviewer: Is there a way to know it's a Vale error vs something else? I'm trying to test it using incorrect words from WordList.yml and I don't think it's working but I can't tell for sure.

Author: If it's working, you get a squiggly line under an error in the text plus something in the PROBLEMS report.
![image](https://github.com/grafana/writers-toolkit/assets/131809008/747ac599-1754-43a3-8e56-105f0f0fcbe0)

> 1. A full report in the `PROBLEMS` tab.

Reviewer: ```suggestion
1. A full report in the **PROBLEMS** tab.
```

Author: Yeah... I had it like that and Prettier had a fit. Trying again.

<!-- vale Grafana.Quotes = YES -->
```

## Use Vale in VS Code

You can use Vale to lint your current document in VS Code.

> 1. Clone the [Technical Documentation](https://github.com/grafana/technical-documentation) repository.

Reviewer: ```suggestion
1. Clone the [Writer's Toolkit](https://github.com/grafana/writers-toolkit/) repository.
```

   ```bash
   git clone git@github.com:grafana/technical-documentation.git
   ```

1. Download and install [Vale](https://github.com/errata-ai/vale/releases).

   {{% admonition type="note" %}}
   Refer to the Linux installation steps at [GitHub Releases](https://vale.sh/docs/vale-cli/installation/#github-releases). Verify that you are downloading the latest build of Vale for Linux.
   {{% /admonition %}}

   {{< code >}}

   ```linux
   wget https://github.com/errata-ai/vale/releases/download/v2.28.0/vale_2.28.0_Linux_64-bit.tar.gz
   mkdir bin && tar -xvzf vale_2.28.0_Linux_64-bit.tar.gz -C bin
   export PATH=./bin:"$PATH"
   ```

   ```macos
   brew install vale
   ```

   {{< /code >}}

1. Create a `vale.ini` file in your home directory or in a working directory with the following contents:

   ```bash

>    StylesPath = /FULL_PATH_TO_REPO/technical-documentation/linters/vale
>    MinAlertLevel = suggestion
> 
>    [*.md]
>    BasedOnStyles = Grafana

Author: ```suggestion
   MinAlertLevel = suggestion
   StylesPath = /FULL_PATH_TO_REPO/writers-toolkit/vale
   [*.md]
   BasedOnStyles = Google, Grafana
   Google.Quotes = NO
   Google.Units = NO
   Google.WordList = NO
   TokenIgnores = (<http[^\n]+>+?)
```

1. Install the [Vale VS Code extension](https://marketplace.visualstudio.com/items?itemName=chrischinchilla.vale-vscode) in VS Code.

   1. Start VS Code.
   1. Press Ctrl+P, paste the following command, and press Enter.

   ```
   ext install ChrisChinchilla.vale-vscode
   ```

1. Configure the Vale VS Code extension.

   1. Press Ctrl+Shift+X or click the **Extensions** icon and select the Vale VS Code extension.
   1. Select the gear icon.
   <!-- vale off -->
   1. Set **Vale › Vale CLI: Config** to the path to your `vale.ini` file. For example, on Linux that could be `/home/USERNAME/vale.ini` and on macOS, that could be `/Users/USERNAME/vale.ini`. The path depends on where you created the `vale.ini` file.
   <!-- vale on -->
   1. Set **Vale › Vale CLI: Path** to the path for the vale executable. For example, in Linux, that could be `/home/USERNAME/bin/vale` and on macOS, that could be `/usr/local/bin/vale`.

1. Restart VS Code.

Vale lints your current document every time you save your changes. The extension reports the linting results in two ways:

> 1. In-line edit marks. You can hover your mouse cursor over the edit marks to view the vale warning or error.

Reviewer: ```suggestion
- In-line edit marks. You can hover your mouse cursor over the edit marks to view the vale warning or error.
```

> 1. A full report in the **PROBLEMS** tab.

Reviewer: ```suggestion
- A full report in the **PROBLEMS** tab. These errors include a reference to the Vale rule.
```

<!-- vale Grafana.Quotes = YES -->
```

## Use Vale in VS Code

You can use Vale to lint your current document in VS Code.

1. Clone the [Writer's Toolkit](https://github.com/grafana/writers-toolkit/) repository.

   ```bash
   git clone git@github.com:grafana/writers-toolkit.git
   ```

1. Download and install [Vale](https://vale.sh/docs/vale-cli/installation/).

   {{% admonition type="note" %}}
   Refer to the Linux installation steps at [GitHub Releases](https://vale.sh/docs/vale-cli/installation/#github-releases). Verify that you are downloading the latest build of Vale for Linux.
   You can use your Linux package manager to install Vale if it is included in your distribution repositories.
   {{% /admonition %}}

   {{< code >}}

   ```linux
   wget https://github.com/errata-ai/vale/releases/download/v2.28.0/vale_2.28.0_Linux_64-bit.tar.gz
   mkdir bin && tar -xvzf vale_2.28.0_Linux_64-bit.tar.gz -C bin
   export PATH=./bin:"$PATH"
   ```

   ```macos

>    brew install vale

Reviewer 2: I think we should probably include a link to https://brew.sh/ in this section as well, people might not be familiar with it.

Author: Added as a Before you begin section

<!-- vale Grafana.Quotes = YES -->
```

## Use Vale in VS Code

You can use Vale to lint your current document in VS Code.

1. Clone the [Writer's Toolkit](https://github.com/grafana/writers-toolkit/) repository.

   ```bash
   git clone git@github.com:grafana/writers-toolkit.git
   ```

> 1. Download and install [Vale](https://github.com/errata-ai/vale/releases).

Reviewer 2: ```suggestion
1. Download and install [Vale](https://vale.sh/docs/vale-cli/installation/).
```

Maybe we can link to the Vale installation page here instead? If people are using `brew` or `choco`, they don't have to go the releases page.

   {{% admonition type="note" %}}
   Refer to the Linux installation steps at [GitHub Releases](https://vale.sh/docs/vale-cli/installation/#github-releases). Verify that you are downloading the latest build of Vale for Linux.
   {{% /admonition %}}

   {{< code >}}

   ```linux
   wget https://github.com/errata-ai/vale/releases/download/v2.28.0/vale_2.28.0_Linux_64-bit.tar.gz
   mkdir bin && tar -xvzf vale_2.28.0_Linux_64-bit.tar.gz -C bin
   export PATH=./bin:"$PATH"
   ```

   ```macos
   brew install vale
   ```

   {{< /code >}}

1. Create a `vale.ini` file in your home directory or in a working directory with the following contents:

   ```bash
   MinAlertLevel = suggestion
   StylesPath = /FULL_PATH_TO_REPO/writers-toolkit/vale
   [*.md]
   BasedOnStyles = Google, Grafana
   Google.Quotes = NO
   Google.Units = NO
   Google.WordList = NO
   TokenIgnores = (<http[^\n]+>+?)
   ```

>    Replace `FULL_PATH_TO_REPO` with the full path to the cloned Technical Documentation repository. For example, in Linux you could set StylesPath to `/home/username/git-repos/technical-documentation/linters/vale` and in macOS, you could set it to `/Users/username/git-repos/technical-documentation/linters/vale`. The path depends on where you cloned the git repository.

Reviewer: ```suggestion
   Replace `FULL_PATH_TO_REPO` with the full path to the cloned Technical Documentation repository. For example, in Linux you could set StylesPath to `/home/username/git-repos/technical-documentation/linters/vale` and in macOS, you could set it to `/Users/username/git-repos/writers-toolkit/vale`. The path depends on where you cloned the git repository.
```

1. Install the [Vale VS Code extension](https://marketplace.visualstudio.com/items?itemName=chrischinchilla.vale-vscode) in VS Code.

   1. Start VS Code.

>    1. Press Ctrl+P, paste the following command, and press Enter.

Reviewer: Wondering if we might also just add how  you can do this in the UI (i.e., Click Extensions, search for Vale VS Code, click Install."

Author: We could... I never use that route, so it never occurred to me to add it :-P

Reviewer: Using the UI always occurs to me, lol. I hate keyboard commands. They're like a weird hidden language to me. I've made a suggestion below.

   ```

   ```macos
   brew install vale
   ```

   {{< /code >}}

1. Create a `vale.ini` file in your home directory or in a working directory with the following contents:

   ```bash
   MinAlertLevel = suggestion
   StylesPath = /FULL_PATH_TO_REPO/writers-toolkit/vale
   [*.md]
   BasedOnStyles = Google, Grafana
   Google.Quotes = NO
   Google.Units = NO
   Google.WordList = NO
   TokenIgnores = (<http[^\n]+>+?), \*\*[^\n]+\*\*
   ```

   Replace `FULL_PATH_TO_REPO` with the full path to the cloned Writer's Toolkit repository. For example, in Linux you could set StylesPath to `/home/username/git-repos/writers-toolkit/vale` and in macOS, you could set it to `/Users/username/git-repos/writers-toolkit/vale`. The path depends on where you cloned the git repository.

1. Install the [Vale VS Code extension](https://marketplace.visualstudio.com/items?itemName=chrischinchilla.vale-vscode) in VS Code.

   1. Start VS Code.
   1. Press Ctrl+P, paste the following command, and press Enter.

   ```
   ext install ChrisChinchilla.vale-vscode
   ```

1. Configure the Vale VS Code extension.

   1. Press Ctrl+Shift+X or click the **Extensions** icon and select the Vale VS Code extension.

>    1. Select the gear icon.
>    <!-- vale off -->
>    1. Set **Vale › Vale CLI: Config** to the path to your `vale.ini` file. For example, on Linux that could be `/home/USERNAME/vale.ini` and on macOS, that could be `/Users/USERNAME/vale.ini`. The path depends on where you created the `vale.ini` file.
>    <!-- vale on -->
>    1. For Linux, set **Vale › Vale CLI: Path** to the path for the vale executable. For example, that could be `/home/USERNAME/bin/vale`.

Author: ```suggestion
   1. Select the gear icon.
   1. Set **Vale › Vale CLI: Config** to the path to your `vale.ini` file. For example, on Linux that could be `/home/USERNAME/vale.ini` and on macOS, that could be `/Users/USERNAME/vale.ini`. The path depends on where you created the `vale.ini` file.
   1. For Linux, set **Vale › Vale CLI: Path** to the path for the vale executable. For example, that could be `/home/USERNAME/bin/vale`.
```

<!-- vale Grafana.Quotes = YES -->
```

> ## Use Vale in VS Code

Reviewer 3: These instructions are only for macOS. Should we call that out in this title? Unless you have additional task for Windows or Linux?

Author: macOS and Linux are in this iteration... Windows... haven't thought about that yet. The steps should be similar... close-ish... I'll test and write up Windows in an update.

You can use Vale to lint your current document in VS Code.

### Before you begin

> Install [Homebrew](https://brew.sh/) on macOS.

Author: ```suggestion
If you are installing on macOS, make sure that [Homebrew](https://brew.sh/) is installed.
```

   {{% /admonition %}}

   {{< code >}}

   ```linux
   wget https://github.com/errata-ai/vale/releases/download/v2.28.0/vale_2.28.0_Linux_64-bit.tar.gz
   mkdir bin && tar -xvzf vale_2.28.0_Linux_64-bit.tar.gz -C bin
   export PATH=./bin:"$PATH"
   ```

   ```macos
   brew install vale
   ```

   {{< /code >}}

1. Create a `vale.ini` file in your home directory or in a working directory with the following contents:

   ```bash
   MinAlertLevel = suggestion
   StylesPath = /FULL_PATH_TO_REPO/writers-toolkit/vale
   [*.md]
   BasedOnStyles = Google, Grafana
   Google.Quotes = NO
   Google.Units = NO
   Google.WordList = NO
   TokenIgnores = (<http[^\n]+>+?), \*\*[^\n]+\*\*
   ```

   Replace `FULL_PATH_TO_REPO` with the full path to the cloned Writer's Toolkit repository. For example, in Linux you could set StylesPath to `/home/username/git-repos/writers-toolkit/vale` and in macOS, you could set it to `/Users/username/git-repos/writers-toolkit/vale`. The path depends on where you cloned the git repository.

1. Install the [Vale VS Code extension](https://marketplace.visualstudio.com/items?itemName=chrischinchilla.vale-vscode) in VS Code.

   1. Start VS Code.

>    1. Press Ctrl+P, paste the following command, and press Enter.

Reviewer: ```suggestion
   1. Press Ctrl+P, paste the following command, and press Enter. Alternatively, click the **Extensions** icon, search for "Vale VS Code", open it, and click "Install".
```
